packages/mapi-client/src/utils/calculate-retry-delay.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Calculates the delay for the next retry attempt using exponential backoff with full jitter.
+ *
+ * @param attempt The current retry attempt number (starting from 0 for the first retry).
+ * @param baseDelay The initial delay in milliseconds (e.g., 100).
+ * @param maxDelay The maximum possible delay in milliseconds (e.g., 20000).
+ * @returns The calculated delay in milliseconds to wait before the next attempt.
+ */
+export function calculateRetryDelay(
```

Comment:
praise: this is really sophisticated solution.

I was reading all about it in https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/ and it's super interesting. Thanks!